File: PokerGame/Card.py
class Card:
   
 def __init__(self, rank: str, suit: str): #This makes the card object
    self.rank = rank
    self.suit = suit

 def get_value(self) -> int: #this creates a method to return the int value of the card 
        rank_values = {
           "2":2, "3":3, "4":4, "5":5,
            "6":6, "7":7, "8":8, "9":9, "10":10,
            "J":11, "Q":12, "K":13, "A":14
        }
        return rank_values.get(self.rank,0)
   
 def get_suit_rank(self) -> int: # creates a method for getting the int value of the suits
        suit_rank = {
             'C':1,
             'D':2,
             'H':3,
             'S':4

        }
        return suit_rank.get(self.suit, 0)

 def get_rank(self) ->str: # gets the rankinf of the card as a string
     return self.rank

 def get_suit(self) ->str: # gets the suit fo the card as a string
     return self.suit

 def __str__(self):  # pretty much the toString like in java to replace the default print whne printed "raw"
    return f"{self.rank}{self.suit}"

File: PokerGame/Card.cs
namespace PokerGame
{
    public class Card
    {
        // Variables for the Card class
        private string rank;
        private char suit;

        // Constructor for Card object
        public Card(string rank, char suit)
        {
            this.rank = rank;
            this.suit = suit;
        }

        // Get the numerical value of the card's rank
        public int GetValue()
        {
            switch (rank)
            {
                case "A": return 1;
                case "2": return 2;
                case "3": return 3;
                case "4": return 4;
                case "5": return 5;
                case "6": return 6;
                case "7": return 7;
                case "8": return 8;
                case "9": return 9;
                case "10": return 10;
                case "J": return 11;
                case "Q": return 12;
                case "K": return 13;
                default: return 0;
            }
        }

        // Get the suit's numerical rank
        public int GetSuitRank()
        {
            switch (suit)
            {
                case 'C': return 1; // Clubs
                case 'D': return 2; // Diamonds
                case 'H': return 3; // Hearts
                case 'S': return 4; // Spades
                default: return 0;
            }
        }

        // Overrides ToString to print the card nicely
        public override string ToString()
        {
            return $"{rank}{suit}";
        }

        // Getters for rank and suit
        public char GetSuit()
        {
            return suit;
        }

        public string GetRank()
        {
            return rank;
        }
    }
}
